src/ucp/wireup/address.c

```diff
@@ -863,13 +876,12 @@ ucs_status_t ucp_address_unpack(ucp_worker_t *worker, const void *buffer,
     }
 
     /* Unpack addresses */
-    address = address_list;
-    ptr     = aptr;
+    address   = address_list;
+    ptr       = aptr;
     dev_index = 0;
-    do {
-        if (*(uint8_t*)ptr == UCP_NULL_RESOURCE) {
-            break;
-        }
+
+    last_dev = (*(uint8_t*)ptr == UCP_NULL_RESOURCE);
```

Comment:
it should not be possible here right? You would `goto out` on line 867

Reply:
yes, in case of NULL_RESOURCE flag is set, it would skip the loop and go to `out`

Reply:
i mean that there is no point to check this already here